pinot-core/src/main/java/org/apache/pinot/core/operator/CombineOperator.java

```diff
@@ -47,22 +45,16 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(CombineOperator.class);
   private static final String OPERATOR_NAME = "CombineOperator";
 
+  // Use at most 10 or half of the processors threads for each query.
+  // If there are less than 2 processors, use 1 thread.
+  // Runtime.getRuntime().availableProcessors() may return value < 2 in container based environment, e.g. Kubernetes.
+  private static final int MAX_NUM_THREADS_PER_QUERY =
+      Math.max(1, Math.min(10, Runtime.getRuntime().availableProcessors() / 2));
```

Comment:
Will `Runtime.getRuntime().availableProcessors()` still work accurately on cloud VM's (say Azure)? Also, what about when there's cgroups limits imposed?

Reply:
That I'm not sure and we need to test that out. Based on Xiang's experience, Kubernetes might return 1 for this, and that is the reason why we ensure this value is at least 1